build-scripts/MSVC.cmake

```diff
@@ -0,0 +1,56 @@
+if (NOT $ENV{DevEnvDir})
+ set(ENV{DevEnvDir} "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE")
+endif()
+set(CMAKE_MAKE_PROGRAM $ENV{DevEnvDir}/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe CACHE PATH "")
+set(CMAKE_CXX_FLAGS_INIT "\
+/MP /utf-8 /bigobj /permissive- /sdl- /FC /Gd /GS- /Gy \
+/wd4068 /wd4146 /wd4819 /wd6237 /wd6319 /wd26444 /wd26451 /wd26495 /WX- /W1 \
+/TP /Zc:forScope /Zc:inline /Zc:wchar_t"
+)
+# /MP     # cl.exe build with multiple processes
+# /utf-8  # set source and execution character sets to UTF-8
+# /bigobj # increase # of sections in object files
+# /permissive- # to allow alternative operators ("and", "or", "not")
+# /sdl-   # disable additional security checks
+# /FC     # full path in compiler messages
+# /Gd     # __cdecl
+# /GS-    # disable buffer security checks
+# /Gy     # Enable Function-Level Linking
```

Comment:
If we don't have it in the VS configuration, it's an oversight, but we should also enable `/GF` string pooling.

Reply:
ffb24a5 refers